### ISSUE TITLE
tests: Turn assets into resources

### ISSUE THREAD

**Issue by fviernau** [oss-review-toolkit/ort#8589]
Assets have been used historically for performance reasons.
Issue with this is:
- If only asset file is changed, test is not re-run since no changed file detected.
- A few code complications, e.g. getAssetFile()

goal: Move assets to resources and simplify code.

**Comment by mnonnenmacher** [comment]
If adding those files to the resources still has negative impact on performance, another option might be to add them as inputs for the test tasks.

**Comment by sschuberth** [comment]
> If adding those files to the resources still has negative impact on performance

I don't think that's the case. The performance issue was about running individual tests from the IDE, where packaging resources into a JAR jut to extract them again would be sub-optimal. But I just verified that printing the path to a resources yields a URL like

```
file:/home/sebastian/Development/GitHub/oss-review-toolkit/ort/utils/common/build/resources/funTest/testpkg.deb
```

when running a test from the IDE or via Gradle. So an optimization is done to access file file from the file system, directly. Otherwise the URL would have been something like `jar:common-utils.jar!/testpkg.deb`.

> another option might be to add them as inputs for the test tasks.

While that would solve the issue with tests not being rerun if only the expected results change, it would lose other benefits of using the conventional `(fun)Test/resources`, like simpler setup with frameworks as WireMock, which use these locations by default.

**Comment by fviernau** [comment]
> it would lose other benefits of using the conventional (fun)Test/resources, like simpler setup with frameworks as WireMock

Do these "other benefits" mainly apply to `test` but not `funTest` ? 

Another maybe related thing: Do we intend to move all these projects used by the fun tests also under resources ?

**Comment by fviernau** [comment]
Thanks!

> But right now, I do not see a need for that.

Have you considered the `external` projects, residing in Git submodules too ?

**Comment by sschuberth** [comment]
> Have you considered the `external` projects, residing in Git submodules too ?

No, not yet. Maybe this is a good time to get rid of them? While on the one hand it's nice and convenient to use real-world test data, updating expected result for them can be a pain, and it's harder to test specific features with them. So maybe we should fully rely on synthetic test projects instead (that may be based off external projects).

**Comment by fviernau** [comment]
Let's also consider analyzer fun tests, which often leave modified files. If the files are under resources, does it cause a rebuild
between running a test multiple times?

**Comment by sschuberth** [comment]
> Let's also consider analyzer fun tests, which often leave modified files.

I'm considering this to be a bug then, BTW.

> If the files are under resources, does it cause a rebuild between running a test multiple times?

Probably yes.

IMO our goal should anyway be to never leave modified files behind. A user who runs an ORT analysis would not want to get a dirty Git working tree just because of that. And I consider it to be good practice to not unsolicitedly create files in the project tree.

**Comment by sschuberth** [comment]
As of https://github.com/oss-review-toolkit/ort/pull/10318, only package managers remain to be migrated from assets to resources.
